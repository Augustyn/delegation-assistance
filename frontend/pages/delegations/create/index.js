import CreateDelegationsPage from "../../../src/features/CreateDelegationPage";
export default CreateDelegationsPage;
