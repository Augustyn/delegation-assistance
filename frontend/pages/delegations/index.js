import DelegationsPage from "../../src/features/DelegationsPage";
export default DelegationsPage;
