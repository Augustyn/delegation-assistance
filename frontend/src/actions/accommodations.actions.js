import { APIService } from "../services/data";

export const ACTIONS = {
  FETCH_ACCOMMODATIONS: "ACCOMMODATIONS_FETCH_ACCOMMODATIONS",
  ADD_ACCOMMODATION: "ACCOMMODATIONS_ADD_ACCOMMODATION",
  SORT_ACCOMMODATIONS: "ACCOMMODATIONS_SORT_ACCOMMODATIONS"
};

const addAccommodation = (delegationId, accommodation) => dispatch => {
  return dispatch(
    APIService.post(ACTIONS.ADD_ACCOMMODATION, {
      url: `/delegations/${delegationId}/accommodations`,
      headers: {
        "Content-type": "application/json"
      },
      needAuth: true,
      data: accommodation
    })
  );
};

const fetchAccommodations = delegationId => dispatch => {
  return dispatch(
    APIService.get(ACTIONS.FETCH_ACCOMMODATIONS, {
      url: `/delegations/${delegationId}/accommodations`,
      headers: {
        "Content-type": "application/json"
      },
      needAuth: true
    })
  );
};

const sortAccommodationsByCheckInDate = order => dispatch => {
  return dispatch({
    type: ACTIONS.SORT_ACCOMMODATIONS,
    payload: order
  });
};

export { addAccommodation, fetchAccommodations, sortAccommodationsByCheckInDate };
