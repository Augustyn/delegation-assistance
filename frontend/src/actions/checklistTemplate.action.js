import { APIService } from "../services/data";

export const ACTIONS = {
  GET_GLOBAL_TEMPLATE: "CHECKLIST_TEMPLATE_GET_GLOBAL_TEMPLATE",
  REORDER_GLOBAL_TEMPLATE: "CHECKLIST_TEMPLATE_REORDER_GLOBAL_TEMPLATE",
  SAVE_GLOBAL_TEMPLATE: "CHECKLIST_TEMPLATE_SAVE_GLOBAL_TEMPLATE",
  ADD_TASK: "CHECKLIST_TEMPLATE_ADD_TASK",
  EDIT_TASK: "CHECKLIST_TEMPLATE_EDIT_TASK",
  DELETE_TASK: "CHECKLIST_TEMPLATE_DELETE_TASK"
};

const fetchGlobalTemplate = () => dispatch => {
  return dispatch(
    APIService.get(ACTIONS.GET_GLOBAL_TEMPLATE, {
      url: "/checklist",
      headers: {
        "Content-type": "application/json"
      },
      needAuth: true
    })
  );
};

const saveGlobalTemplate = globalTemplate => dispatch => {
  return dispatch(
    APIService.put(ACTIONS.SAVE_GLOBAL_TEMPLATE, {
      url: "/checklist",
      headers: {
        "Content-type": "application/json"
      },
      needAuth: true,
      data: {
        activities: [...globalTemplate]
      }
    })
  );
};

const reorderGlobalTemplate = ({ oldIndex, newIndex }) => dispatch =>
  dispatch({
    type: ACTIONS.REORDER_GLOBAL_TEMPLATE,
    payload: {
      oldIndex: oldIndex,
      newIndex: newIndex
    }
  });

const addTask = ({ task, description }) => dispatch =>
  dispatch({
    type: ACTIONS.ADD_TASK,
    payload: {
      task: task,
      description: description
    }
  });

const editTask = ({ id, version, task, description, priority }) => dispatch =>
  dispatch({
    type: ACTIONS.EDIT_TASK,
    payload: {
      id: id,
      version: version,
      task: task,
      description: description,
      priority: priority
    }
  });

const deleteTask = index => dispatch =>
  dispatch({
    type: ACTIONS.DELETE_TASK,
    payload: {
      index: index
    }
  });

export { fetchGlobalTemplate, reorderGlobalTemplate, saveGlobalTemplate, addTask, editTask, deleteTask };
