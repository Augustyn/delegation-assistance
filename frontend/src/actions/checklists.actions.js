import { APIService } from "../services/data";

export const ACTIONS = {
  ADD_TASK: "CHECKLISTS_ADD_TASK",
  GET_TASKS: "CHECKLISTS_GET_TASKS",
  DELETE_TASK: "CHECKLISTS_DELETE_TASK"
};

const addTask = (name, description) => dispatch => {
  return dispatch(
    APIService.post(
      ACTIONS.ADD_TASK,
      {
        url: "/checklist/tasks",
        headers: {
          "Content-type": "application/json"
        },
        needAuth: true,
        data: {
          name: name,
          description: description
        }
      },
      {
        name: name,
        description: description
      }
    )
  );
};

const deleteTask = id => dispatch => {
  let url = "/checklist/tasks/" + id;
  return dispatch(
    APIService.delete(
      ACTIONS.DELETE_TASK,
      {
        url: url,
        headers: {
          "Content-type": "application/json"
        },
        needAuth: true
      },
      {
        id: id
      }
    )
  );
};

const fetchTasks = () => dispatch => {
  return dispatch(
    APIService.get(ACTIONS.GET_TASKS, {
      url: "/checklist",
      headers: {
        "Content-type": "application/json"
      },
      needAuth: true
    })
  );
};

const saveTasks = tasks => dispatch => {
  return dispatch(
    APIService.put(ACTIONS.GET_TASKS, {
      url: "/checklist",
      headers: {
        "Content-type": "application/json"
      },
      needAuth: true,
      data: {
        tasks: tasks
      }
    })
  );
};

export { addTask, deleteTask, fetchTasks, saveTasks };
