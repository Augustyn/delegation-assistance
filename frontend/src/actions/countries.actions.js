import { APIService } from "../services/data";

export const ACTIONS = {
  FETCH_COUNTRIES: "COUNTRIES_FETCH_COUNTRIES"
};

const fetchCountries = () => (dispatch, getState) => {
  // Fetching only when coutries collection is empty in redux store.
  const countries = getState().countries.countries;
  if (countries.length === 0) {
    return dispatch(
      APIService.get(ACTIONS.FETCH_COUNTRIES, {
        url: `/countries`,
        headers: {
          "Content-type": "application/json"
        },
        needAuth: true
      })
    );
  }
  return Promise.resolve();
};

export { fetchCountries };
