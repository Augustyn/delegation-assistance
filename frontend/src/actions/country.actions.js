import { RESTCountries } from "../services/data";

export const ACTIONS = {
  FETCH_COUNTRY: "COUNTRIES_FETCH_COUNTRY"
};

const fetchCountry = countryCode => dispatch => {
  return dispatch(
    RESTCountries.get(ACTIONS.FETCH_COUNTRY, {
      url: `/alpha/${countryCode}`,
      headers: {
        "Content-type": "application/json"
      }
    })
  );
};

export { fetchCountry };
