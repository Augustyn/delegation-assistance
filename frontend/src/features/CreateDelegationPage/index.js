import CreateDelegationPage from "./CreateDelegationPage.container";
export default CreateDelegationPage;
