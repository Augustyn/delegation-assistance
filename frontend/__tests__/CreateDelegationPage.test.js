import React from "react";
import Enzyme, { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import configureStore from "redux-mock-store"; 
import thunk from "redux-thunk";

import { CreateDelegationPage } from "../src/features/CreateDelegationPage/CreateDelegationPage.component";
import { addNewDelegation } from "../src/actions/delegations.actions";

const middlewares = [thunk]; 
const mockStore = configureStore(middlewares);

Enzyme.configure({ adapter: new Adapter() });

const setup = () => {
  const props = {
    handleSubmit: jest.fn(),
    subbmiting: false,
    invalid: false
  };

  const enzymeWrapper = shallow(<CreateDelegationPage {...props} />);

  return {
    props,
    enzymeWrapper
  };
};

describe("Create Delegation Page", () => {

  let enzymeWrapper;
  let props;
  
  beforeEach(() => {
     ({ enzymeWrapper, props } = setup());
  });

  it("should render correctly", () => {
    expect(enzymeWrapper).toMatchSnapshot();
  });

  it("should call submit with delegation object", () => {
    const value = {
      startDate: "02-03-2004",
      endDate: "02-03-2004",
      delegationObjective: "testObjective",
      destinationCountryISO3: "testCountry",
      destinationLocation: "testLocation"
    };
    enzymeWrapper.find("Form").simulate("submit", value);
    expect(props.handleSubmit).toBeCalledWith(value);
  });

  it("should render 3 Inputs", () => {
    expect(enzymeWrapper.find("Input").length).toBe(3);
  });

  it("should render destinationCountryISO3 Input", () => {
    expect(
      enzymeWrapper.find("Input[name='destinationCountryISO3']").length
    ).toBe(1);
  });

  it("should render destinationLocation Input", () => {
    expect(enzymeWrapper.find("Input[name='destinationLocation']").length).toBe(
      1
    );
  });

  it("should render delegationObjective Input", () => {
    expect(enzymeWrapper.find("Input[name='delegationObjective']").length).toBe(
      1
    );
  });

  it("should render startDate DatePicker", () => {
    expect(enzymeWrapper.find("Field[name='startDate']").length).toBe(1);
  });

  it("should render endDate DatePicker", () => {
    expect(enzymeWrapper.find("Field[name='endDate']").length).toBe(1);
  });

  it("should dispatch action", () => {
    const initialState = {};
    const store = mockStore(initialState);
    store.dispatch(addNewDelegation());

    const actions = store.getActions();
    const expectedPayload = "DELEGATIONS_ADD_DELEGATION";
    expect(actions[0].type).toEqual(expectedPayload);
  });
});
