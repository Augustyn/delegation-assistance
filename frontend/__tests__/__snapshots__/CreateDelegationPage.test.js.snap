// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create Delegation Page should render correctly 1`] = `
<Form
  id="create-delegation"
  onSubmit={[MockFunction]}
>
  <div
    className="container"
  >
    <Input
      label="Destination Country"
      name="destinationCountryISO3"
      placeholder="Destination Country"
      validate={[Function]}
    />
    <Input
      label="Destination Location"
      name="destinationLocation"
      placeholder="Destination Location"
      validate={[Function]}
    />
    <Input
      label="Delegation Objective"
      name="delegationObjective"
      placeholder="Delegation Objective"
      validate={[Function]}
    />
    <div
      className="date-pickers-container"
    >
      <div>
        <label
          htmlFor="startDate"
        >
          Start date:
        </label>
        <Field
          component={[Function]}
          name="startDate"
        />
      </div>
      <div>
        <label
          htmlFor="endDate"
        >
          End date:
        </label>
        <Field
          component={[Function]}
          name="endDate"
        />
      </div>
    </div>
    <button
      className="btn btn-primary btn-create-delegation"
      type="submit"
    >
      CREATE DELEGATION
    </button>
  </div>
</Form>
`;
